src/System.Memory/ref/System.Memory.cs

```diff
@@ -87,6 +87,7 @@ public static class SpanExtensions
     {
          public static int IndexOf<T>(this Span<T> span, T value) where T:struct, IEquatable<T> { throw null; }
```

Comment:
Wow, I look forward to seeing this. @ahsonkhan Can you make sure we have a benchmark for this scenario? It's pretty fundamental.